services/motion/builtin/move_request.go

```diff
@@ -304,6 +291,22 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
 			inputMap[mr.kinematicBase.Name().ShortName()] = currentInputs
 
+			// get the current position of the base
+			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+			if err != nil {
+				return state.ExecuteResponse{}, err
+			}
+
+			// Note: the value of wayPointIndex is subject to change between when this function is first entered
+			// versus when CheckPlan is actually called.
+			// We load the wayPointIndex value to ensure that all information is up to date.
+			waypointIndex := int(mr.waypointIndex.Load())
```

Comment:
This seems like a race. If execution swaps to the next trajectory in between getting `currentInputs` and getting `waypointIndex` then these will not match.

I think we need a mutex protecting this so that `currentInputs` and `waypointIndex` can be guaranteed to match.